### DISCUSSION TITLE
Translating into other languages.

### DISCUSSION THREAD

**Discussion by meteos77** [textbrowser/glitch#4] [Ideas]
I hope there will be the possibility to translate this software in the language of your choice :-)

**Comment by textbrowser** [comment]
Translating it like whenever you wish? Or from the system?

**Reply by meteos77** [reply]
I do not understand your question but my question was not precise enough
I was thinking about the translation with linguist of the application menu like BiblioteQ or Dooble

**Comment by textbrowser** [comment]
BiblioteQ is the odd one because dynamic translations require a lot of work whereas static translations are natural. A static translation is one in which the application is translated during launch based on the system's locale. A dynamic translation requires menus and widgets that automatically adjust themselves. I prefer static.

**Reply by meteos77** [reply]
for the dynamic or static user does not change anything?

**Comment by textbrowser** [comment]
Dynamic: user controls.
Static: software.

**Comment by meteos77** [comment]
sorry but I don't understand the nuance
if I do the command in the Sources directory : cat *  | grep tr\(     =>  I get 127 lines

at what point should the translation be done in static to have the different texts framed by tr("english") and become "french" for the user?

**Comment by textbrowser** [comment]
https://doc.qt.io/qt-5/internationalization.html#dynamic-translation

**Comment by meteos77** [comment]
thank you for your explanations, I think I understood the nuance
BiblioteQ is lucky to be dynamic because if you change language you don't need to restart the application.
whereas Dooble, put the parameter "system". 

the dynamic system is the one used for example on the automat of sale of ticket of train, the 1st menu asks to choose the language and indeed it does not restart the application :-)

**Reply by textbrowser** [reply]
The effort is exponential or polynomial for dynamic translations.

**Reply by textbrowser** [reply]
For a kiosk application, dynamic is appropriate. For a desktop, probably not. I doubt Dooble is appearing in kiosks.

**Comment by meteos77** [comment]
I've started translating glitch but I can't see in the software what it looks like. It doesn't offer French despite the .qm in a Translations directory.

Not being a developer, the translation may not be optimal.

Can you make sure that I can see the French in the application itself. that will facilitate the correction of ""bug""

thank you in advance

[2022-12-01_Packages-French-initial.tar.gz](https://github.com/textbrowser/glitch/files/10133116/2022-12-01_Packages-French-initial.tar.gz)

**Reply by textbrowser** [reply]
Thanks you very much. This will be included in the future.

**Reply by textbrowser** [reply]
https://github.com/textbrowser/glitch/discussions/6

Looks like that but newer.

**Comment by textbrowser** [comment]
There's more work in the languages department. It'll include live language options like in BiblioteQ. It's not easy and boring. But there will always be someone asking for their native language without having to restart the program. Then there's the embedding of the files into the application. I'm going to replace all written languages with a universal black and white Emoji language.

**Reply by meteos77** [reply]
Restarting the program once to be in its language is not a problem.

The program is set to English by default but if you set it to system language by default, it should adapt itself?

**Reply by meteos77** [reply]
>  I'm going to replace all written languages with a universal black and white Emoji language.

Very good idea but you need a lot of drawing talent :-)

On the other hand you will have a huge potential of users : https://en.wikipedia.org/wiki/List_of_countries_by_literacy_rate
and the literate amateurs
